### PR TITLE
62 Implement DataAccess Layer

### LINKED ISSUE #62: Implement DataAccess Layer
Create the data access layer for aFFR Service
Based on Figma Artifacts: Relational DB Schema, Architecture and Domain models
https://www.figma.com/file/k3bsKz9Y2G8l2z0FD4D0iY/System-Development-Artifacts?node-id=0%3A1&t=JZskpZTIs81V5l8p-0

### PR DESCRIPTION
**Pull request by AlexStavrev**
Add a Data Access Layer to the project. It handles CRUD for Bid and Signal models.

### CODE REVIEW

**Comment by WrexBG** [comment]
Also waiting on adding the workflow to be merged to [main]
 * #70

**Comment by WrexBG** [comment]
Needs a version tag before being merged as well.

File: aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs
Comment (mette-thorhauge) [inline aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs:101]: Make sure that we assert that the updated price is not equal to the old price.

File: aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs
Comment (mette-thorhauge) [inline aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs:162]:
```suggestion
        //Assert
```

File: aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs
Comment (mette-thorhauge) [inline aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs:132]:
```suggestion
        //Assert
```

File: aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs
Comment (mette-thorhauge) [inline aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs:118]:
```suggestion
        //Assert
```

File: aFRR-Service/BaseDataAccess/BaseDataAccess.cs
Comment (mette-thorhauge) [inline aFRR-Service/BaseDataAccess/BaseDataAccess.cs:176]: Consider whether it should be static

**Review by mette-thorhauge — CHANGES_REQUESTED** [review]
Review has been finished by me and @zythaarcz, different comments have been made, most address minor changes.

File: aFRR-Service/BaseDataAccess/BaseDataAccess.cs
Comment (WrexBG) [reply aFRR-Service/BaseDataAccess/BaseDataAccess.cs]: Approved with: Commit https://github.com/diversiwaver/aFRR-NET/commit/86896265837859af1651e17dc3f24cffbdd48572

File: aFRR-Service/BaseDataAccess/BaseDataAccess.cs
Comment (WrexBG) [reply aFRR-Service/BaseDataAccess/BaseDataAccess.cs:35]: Approved with: Commit https://github.com/diversiwaver/aFRR-NET/commit/7312bb057604c53fa86797f7889c0207ee608370

File: aFRR-Service/TestDataAccess/Tests/TestBidDataAccess.cs
Comment (WrexBG) [reply aFRR-Service/TestDataAccess/Tests/TestBidDataAccess.cs]: Approved with: Commit b93980745c1d0b8fa72d371fb71f5de51b0dc2c8

File: aFRR-Service/TestDataAccess/Tests/TestBidDataAccess.cs
Comment (WrexBG) [reply aFRR-Service/TestDataAccess/Tests/TestBidDataAccess.cs:103]: Approved with: Commit 4ffb0403b2afcc1a0f01a828f819fa0e1060a99f

File: aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs
Comment (WrexBG) [reply aFRR-Service/TestDataAccess/Tests/TestSignalDataAccess.cs:104]: Approved with: Commit 4ffb0403b2afcc1a0f01a828f819fa0e1060a99f

**Review by mette-thorhauge — COMMENTED** [review]
All previous concerns have been changed, so I have no further comments.

**Comment by WrexBG** [comment]
Rebased the branch from main to resolve the merge conflicts

**Review by mette-thorhauge — APPROVED** [review]
The final changes have been made and approved

**Pull request merged**
